src/ckeditor.tsx

```diff
@@ -239,26 +299,105 @@ export default class CKEditor<TEditor extends Editor> extends React.Component<Pr
 		} );
 	}
 
+	private getStateDiff( previousState: Record<string, unknown>, newState: Record<string, unknown> ):
+		{ addedKeys: Array<string>; removedKeys: Array<string>; modifiedKeys: Array<string> } {
+		const previousStateKeys = Object.keys( previousState );
+		const newStateKeys = Object.keys( newState );
+
+		return {
+			addedKeys: newStateKeys.filter( key => previousStateKeys.indexOf( key ) === -1 ),
+			removedKeys: previousStateKeys.filter( key => newStateKeys.indexOf( key ) === -1 ),
+			modifiedKeys: previousStateKeys.filter( key => newStateKeys.indexOf( key ) !== -1 &&
+				JSON.stringify( previousState[ key ] ) !== JSON.stringify( newState[ key ] ) )
+		};
+	}
+
+	private _handleRoots = (
```

Comment:
Why this is not a class method?

Reply:
This is a method of the CKeditor react integration class. Do you mean it should be implemented inside the `MultiRootEditor`?

Reply:
I added a comment with the suggested change: https://github.com/ckeditor/ckeditor5-react/pull/390#discussion_r1258624209